camel-idea-plugin/src/main/java/com/github/cameltooling/idea/preference/annotator/CamelAnnotatorPage.java

```diff
@@ -50,6 +51,7 @@ public JComponent createComponent() {
         highlightCustomOptionsCheckBox = new JBCheckBox("Highlight custom endpoint options as warnings in editor");
         realTimeSimpleValidationCatalogCheckBox = new JBCheckBox("Real time validation of Camel simple language in editor");
         realTimeJSonPathValidationCatalogCheckBox = new JBCheckBox("Real time validation of Camel JSonPath language in editor");
+        realTimeIdReferenceTypeValidationCheckBox = new JBCheckBox("Real time type validation when referencing beans");
```

Comment:
This only true for XML bean reference, right ?

Reply:
Anywhere where a bean reference is used (so far the only place other than XML is @BeanInject(value = "HERE") - code completion will only offer appropriate beans, and if someone references a bean with different type, it will be annotated.